mlflow/pyfunc/__init__.py

```diff
@@ -422,9 +422,12 @@ def spark_udf(spark, model_uri, result_type="double"):
     def predict(*args):
         model = SparkModelCache.get_or_load(archive_path)
         schema = {str(i): arg for i, arg in enumerate(args)}
-        # Explicitly pass order of columns to avoid lexicographic ordering (i.e., 10 < 2)
-        columns = [str(i) for i, _ in enumerate(args)]
-        pdf = pandas.DataFrame(schema, columns=columns)
+        if type(args[0]) == pandas.DataFrame:
```

Comment:
Hm what about the other args (columns)? Are we assuming that if a user calls the UDF on a StructType column, that (StructType) column is the only column supplied to the UDF?